k3sAndProviderServices.sh

```diff
@@ -56,6 +58,12 @@ while getopts ":d:e:tagm:c:r:w:n:s:" opt; do
       echo "Invalid option: $OPTARG" 1>&2
       exit 1
       ;;
+    k )
+      kubelet_storage="--kubelet-arg=root-dir=$OPTARG"
```

Comment:
We can pass not only the `root-dir` argument using `--kubelet-arg`, but also other options if needed. 

We could address this either in the current PR or in a future update, should we need to configure additional kubelet options via `--kubelet-arg` beyond just `root-dir`.

How this could be addressed is:

1. move line 18 after line 73 (after `shift $((OPTIND -1))`) and extend it with `${kubelet_args}`
```
kubelet_args=""
k3s_common_args="--disable=${disable_components} --flannel-backend=none --kubelet-arg=${kubelet_args}"
```

2. for `k )` argument, use newly introduced `kubelet_args`
```
    k )
      kubelet_args="--root-dir=$OPTARG"
```

3. Remove `$kubelet_storage` from line 185 since `kubelet_args` with `--root-dir` will come via `k3s_common_args`, e.g.:
```
    curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="$k3s_common_args $install_exec $containerd_storage" sh -
```

And leave `$containerd_storage` as-is.

Reply:
@shimpa1  since it requires more work and we need to build new providers rather ASAP now -- we can tackle this comment later in a separate PR (should then create a new issue from [this](https://github.com/chainzero/provider-build-scripts/pull/19/files#r1877892003) comment).

Reply:
I agree.
Both K3S build scripts can be used as they are right now and we can update them later.